packages/table/src/Row.tsx

```diff
@@ -0,0 +1,312 @@
+
+    & > td > div {
+      max-height: 40px;
+    }
+  `,
+};
+
+const styleMap = {
+  left: [DataType.String, DataType.Weight, DataType.ZipCode, DataType.Date],
+  right: [DataType.Number, DataType.Quantity],
+};
+
+function styleColumn(index: string, dataType: DataType) {
+  let justify = '';
+
+  if (styleMap.left.includes(dataType)) {
+    justify = 'flex-start';
+  } else if (styleMap.right.includes(dataType)) {
+    justify = 'flex-end';
+  }
+
+  return css`
+    & td:nth-child(${index}) > div {
+      justify-content: ${justify};
+    }
+  `;
+}
+
+function getIndentLevelStyle(indentLevel: number) {
+  return css`
+    & > td:nth-child(2) {
+      padding-left: ${8 + indentLevel * 16}px;
+    }
+  `;
+}
+
+interface RowProps extends React.ComponentPropsWithoutRef<'tr'> {
+  expanded?: boolean;
+  disabled?: boolean;
+  indentLevel?: number;
+  isParentExpanded?: boolean;
+}
+
+const Row = React.forwardRef(
+  (
+    {
+      expanded = false,
+      disabled = false,
+      children,
+      className,
+      indentLevel = 0,
+      isParentExpanded = true,
+      ...rest
+    }: RowProps,
+    ref: React.Ref<any>,
+  ) => {
+    const {
+      state: { data, columnInfo, hasNestedRows, hasRowSpan, selectable },
+      dispatch,
+    } = useTableContext();
+
+    const indexRef = React.useRef(Math.floor(Math.random() * 1000000 + 1));
+
+    const [isExpanded, setIsExpanded] = useState(expanded);
+    const nodeRef = React.useRef(null);
+    let hasSeenFirstCell = false;
+
+    useEffect(() => {
+      let hasDispatchedHasNestedRows = false;
+      let hasDispatchedHasRowSpan = false;
+
+      React.Children.forEach(children, child => {
+        if (
+          isComponentType(child, 'Row') &&
+          !hasDispatchedHasNestedRows &&
+          !hasNestedRows
+        ) {
+          dispatch({
+            type: Types.SetHasNestedRows,
+            payload: true,
+          });
+
+          hasDispatchedHasNestedRows = true;
+        }
+
+        if (
+          isComponentType(child, 'Cell') &&
+          child.props.rowSpan > 1 &&
+          !hasRowSpan &&
+          !hasDispatchedHasRowSpan
+        ) {
+          dispatch({
+            type: Types.SetHasRowSpan,
+            payload: true,
+          });
+
+          hasDispatchedHasRowSpan = true;
+        }
+      });
+    }, [children]);
+
+    // Depending on network speed, will noticeably render columns with incorrect
+    // alignment, would rather wait for proper information before rendering
+    if (!columnInfo) {
+      return null;
+    }
+
+    const chevronButton = (
+      <IconButton
+        onClick={() => setIsExpanded(curr => !curr)}
+        aria-label="chevron"
+        className={iconButtonMargin}
+      >
+        <Icon
+          aria-label="chevron"
+          glyph={isExpanded ? 'ChevronDown' : 'ChevronRight'}
+          color={uiColors.gray.dark2}
+        />
+      </IconButton>
+    );
+
+    const nestedRows = React.Children.map(children, child => {
```

Comment:
Can we merge this into the below map so we only iterate over children once?

Reply:
Struggling to combine these because the second iteration relies on the updated children from the first map.

Reply:
Combined !